### DISCUSSION TITLE
Testing power monitor with 1 CT on 1 wire of a dryer cord

### DISCUSSION THREAD

**Discussion by acabrera2012** [David00/rpi-power-monitor#146] [Q&A] [answered]
I want to make sure I understand the data that I'm seeing coming from the Power Monitor. I have a 400A CT that I have clamped onto one of my dryer wires coming out of the 240v outlet it is connected to. After following the calibration steps from the docs, I am a bit confused about the config settings and the output I am seeing.

In the calibration instructions from this link: [https://david00.github.io/rpi-power-monitor/docs/v0.3.0/calibration](https://david00.github.io/rpi-power-monitor/docs/v0.3.0/calibration), it mentions using a multi-meter for calibrating the output of the outlet that is being used for the 9V AC adapter. The outlet is a 120v as opposed to the 240v used for the dryer. But, we are placing these settings in the config section labeled [grid_voltage]. How does the outlet used for the 9V AC adapter correlate to the settings needed for the CT? Shouldn't the grid_voltage setting in the config be 240v instead of 120v for the adapter? And, why does the ac_transformer_output_voltage matter in the [grid_voltage] config if what I really need is an accurate current reading from one of the dryer wires coming out of the 240v outlet?

Even so, when I change the config to accommodate my 240v needs, the current readings I see for the dryer are about 10 times higher than what my multi-meter reports for the dryer wire. If I modify the calibration setting of the CT to 0.10, the current reading looks correct.

To ask my question another way, if I just want to test the Power Monitor on a simple dryer wire instead of the electrical panel, what is the correct way to do it?

**Comment by David00** [comment]
Hi @acabrera2012, great question!

Your 240V circuit for the electric dryer is using a 2-pole breaker to take advantage of both 120V legs of your "split phase" electrical service.  Essentially, your power is fed from a 240V circuit from the utility, but it's split into two, and you end up with three possible electrical connections within your home:
 - a 120V outlet to the first half of the power feed
 - a 120V outlet to the second half of the power feed
 - a 240V outlet that takes advantage of the entire feed

Since the majority of outlets in our homes are at 120V, you'll most often run into circuits that align to either option 1 or option 2 from the list above.  Likewise, the 9V AC transformer - which the RPI power monitor uses to measure the voltage and phase - will be connecting to one of the two 120V options.

The power monitor can account for measuring power from 240V circuits by using the [`two_pole`](https://david00.github.io/rpi-power-monitor/docs/v0.3.0/configuration.html#two_pole) configuration option for the CT channel that is clamped over the 240V conductor.

The voltage configuration for the power monitor should be the ~120V measurement coming from the outlet where the Power Monitor's 9V AC transformer is connected.

The `two_pole` configuration option will essentially double the power measurement of the CT that it's applied to. This works fine in most cases, but if your two 120V legs are severely imbalanced (eg, one leg's voltage being at 115V and the other being at 123V), there will be an error in the power measurement of two_pole appliances. However, typically, the two 120V legs are not that far apart, and this ends up working well.

So, in summary:
* Set the power monitor's grid voltage to the ~120V reading at the outlet where it's connected (and set the `ac_transformer_output_voltage` as well, usually around 10-11V AC)
* Set `two_pole = true` for the channel in your config.toml
* Restart the power monitor so the new config values get loaded.

Unless you're using a sensor from my shop, it is a good idea to check that the raw sample data coming in from the sensor looks okay by generating a debug plot.

See "plot" mode [here](https://david00.github.io/rpi-power-monitor/docs/v0.3.0/advanced-usage.html#--mode), and make sure the power monitor isn't [running the background](https://david00.github.io/rpi-power-monitor/docs/v0.3.0/advanced-usage.html#controlling-the-power-monitor-background-service) before using the command line interface.

There are 2 things you'll want to look for in the plot:
* that the CT data for the channel of interest isn't hitting the max value of 1023 on the left axis (the right axis is for the voltage measurements).
* that the CT data is coming in with suitable amplitude (the peak values should ideally be above 700 when the appliance is in use to get good data).

**Reply by acabrera2012** [reply]
Thanks for the input. I have made the changes you mentioned and looked at the output again. While the "amps" readings are more stable now, they are still about 10 times what I would expect now that I changed the calibration back to 1.0. Here is a section of the config in case that helps as well as a screenshot of the output and the plot data:

`[general]
name = "default"
three_phase_mode = false

[database]
host = "localhost"
port = 8086
username = "root"
password = "password"
database_name = "power_monitor"

[grid_voltage]
#grid_voltage = 123.6
grid_voltage = 122.6
#ac_transformer_output_voltage = 10.76
ac_transformer_output_voltage = 10.52
frequency = 60
voltage_calibration = 1

[current_transformers.channel_1]
name = 'My CT Channel'
rating = 400
type = 'mains'
two_pole = true
enabled = true
calibration = 1.0
watts_cutoff_threshold = 1
reversed = true
phase_angle = 0

[current_transformers.channel_2]
name = 'Channel 2'
rating = 100
type = 'mains'
two_pole = false
enabled = true
calibration = 1.0
watts_cutoff_threshold = 0
reversed = false
phase_angle = 0

[current_transformers.channel_3]
name = 'Channel 3'
rating = 100
type = 'consumption'
two_pole = false
enabled = true
calibration = 1.0
watts_cutoff_threshold = 0
reversed = false
phase_angle = 0

[current_transformers.channel_4]
name = 'Channel 4'
rating = 100
type = 'consumption'
two_pole = false
enabled = true
calibration = 1.0
watts_cutoff_threshold = 0
reversed = false
phase_angle = 0

[current_transformers.channel_5]`

I'm sure I must have fat-fingered something but let me know if you have any thoughts. Thanks
<img width="1146" alt="Screenshot 2024-11-25 at 12 44 14 PM" src="https://github.com/user-attachments/assets/b06d88dc-efc1-45f2-8988-898006da931b">
![IMG_8027](https://github.com/user-attachments/assets/167a65ec-54bd-4b67-9c5a-7a5b906dd8ea)

**Reply by David00** [reply, marked as answer]
Everything looks correct from what I can see. I think it must be that the CT you're using is spec'd differently than the ones from my shop.  That's okay, as long as the output from the CT doesn't overload the input of the power monitor, and based on the raw data plot, it's not.  (It looks like the CT1 data peaks around 850 or so which is perfectly fine. because the limit is 1023).

You'll just have to update the calibration value for the channel to get the measurements to align with a handheld AC clamp meter. If they're currently 10x what you are expecting, start with a calibration value of 0.10 for CT1.  Keep in mind you don't have to guess about the calibration value - the docs show a way to divide one reading by the other to get the optimal calibration value (just click to expand the example in the Amperage Calibration section)

**Reply by acabrera2012** [reply]
I understand. Thanks.

**Reply by David00** [reply]
Anytime! Feel free to continue the conversation here if you have any more questions. :)
